### PR TITLE
Enable JitPack publication for Codyze and CoKo

### PR DESCRIPTION
**Pull request by fwendland**
- Update JDK used by JitPack
- Enable and configure Maven publication for Codyze Core, Codyze Backend CPG, Codyze Specification Language CoKo Core Library
- Include sources and docs in publication

Possible improvements open for discussions:
- Custom plugin for publication preventing a lot of copy&paste
- Better way of building and publishing docs with Dokka
- Additional properties for inclusion in POM

Closes: #554

### CODE REVIEW

**Review by maximilian-galanis — CHANGES_REQUESTED** [review]
Please create a publishing convention plugin to reduce the code duplication

File: codyze-backends/cpg/build.gradle.kts [hunk 1]
```diff
@@ -13,3 +14,28 @@ dependencies {
...
+                }
+                scm {
+                    connection.set("scm:git:git://github.com/Fraunhofer-AISEC/codyze.git")
+                    developerConnection.set("scm:git:ssh://git@github.com/Fraunhofer-AISEC/codyze.git")
+                    url.set("https://github.com/Fraunhofer-AISEC/codyze")
+                }
+            }
+        }
+    }
+}
```
Comment (maximilian-galanis) [inline codyze-backends/cpg/build.gradle.kts:41]: Similarly to the [common-conventions plugin](https://github.com/Fraunhofer-AISEC/cpg/blob/main/buildSrc/src/main/kotlin/cpg.common-conventions.gradle.kts#L53) in the CPG project, we could get rid of most of the code duplication here.
In the modules that should be published, this prepared publication can be used like [so](https://github.com/Fraunhofer-AISEC/cpg/blob/main/cpg-analysis/build.gradle.kts#L31).

**Review by maximilian-galanis — CHANGES_REQUESTED** [review]

File: codyze-backends/cpg/build.gradle.kts (hunk 1 above)
Comment (fwendland) [reply codyze-backends/cpg/build.gradle.kts]: I've created a default publication configuration that is adjusted in each module. However, the current names including the artifacts may collide with other packages, e.g., our Codyze backend for the CPG creates the artifact `cpg-.*` under `de.fraunhofer.aisec`.

Do you know a simple way to use, for example, the full module path as publication name (e.g., `codyze-backends:cpg`)?

**Review by maximilian-galanis — APPROVED** [review]
Looks good as far as I can tell.

**Pull request merged**